internal/app/machined/pkg/controllers/k8s/control_plane_static_pod.go

```diff
@@ -423,12 +423,23 @@ func (ctrl *ControlPlaneStaticPodController) manageAPIServer(ctx context.Context
 								v1.ResourceMemory: apiresource.MustParse("512Mi"),
 							},
 						},
+						SecurityContext: &v1.SecurityContext{
+							AllowPrivilegeEscalation: pointer.To(false),
+							Capabilities: &v1.Capabilities{
+								Drop: []v1.Capability{"ALL"},
+								// kube-apiserver binary has cap_net_bind_service=+ep set.
```

Comment:
I remember seeing a fix in k8s/container+runc where the `cap_net_bind_service=+ep` when set on the binary is honored and there is not need to explicitly set `NET_BIND_SERVICE`.

Reply:
okay, that was not a fix, it was a KEP, [KEP-2763](https://github.com/kubernetes/enhancements/pull/2757/), it's  strange that this issue still persists

Reply:
Please tell me where I did go wrong :)
I've created a build without the `Add: [ NET_BIND_SERVICE ]`

```
$ talosctl cluster create --image  ghcr.io/nberlee/talos:v1.1.0-alpha.2-155-ga64941c76-dirty  --wait
validating CIDR and reserving IPs
generating PKI and tokens
downloading ghcr.io/nberlee/talos:v1.1.0-alpha.2-155-ga64941c76-dirty
creating network talos-default
creating master nodes
creating worker nodes
renamed talosconfig context "talos-default" -> "talos-default-8"
waiting for API
bootstrapping cluster
waiting for etcd to be healthy: OK
waiting for etcd members to be consistent across nodes: OK
waiting for apid to be ready: OK
waiting for kubelet to be healthy: OK
◱ waiting for all nodes to finish boot sequence: nodes ["10.5.0.2"] are still in boot sequence
context deadline exceeded
$ talosctl read -n 10.5.0.2 /var/log/pods/kube-system_kube-apiserver-talos-default-master-1_1c0b00f83b9263e8c3da34656ecec18b/kube-apiserver/2.log
2022-07-17T17:11:37.972419441Z stderr F exec /usr/local/bin/kube-apiserver: operation not permitted
```

Reply:
sorry if I confused you, i just linked an old KEP that I've seen, seems it's not completely fixed as i understand. Just a general observation, would have been nice to have it honor ambient capabilities.

Reply:
As I understand  KEP-2763 made ambient capabilities work. And made things worse in this case.
 In this case, we do not care if its working. We actually do not want to honor the filecap, (as we do not need it)
 
 But because ambient capabilities (ie filecap) are honored, and was not impemented as a seperate `ambientCapabilities:`, we now have to allow  `NET_BIND_SERVICE`.

See the example which was previous behavior we would like in our case after `Now the if we update the image and create the pod:-` in [KEP-2763](https://github.com/vinayakankugoyal/enhancements/blob/0cfb55c5b6dd4a0b511fd17701d86a778817b455/keps/sig-security/2763-ambient-capabilities/README.md#motivation)

Reply:
yup, makes sense, I wrongly assumed k8s already implemented setting the ambient capabilities by now